dafni_cli/tests/commands/test_download.py

```diff
@@ -30,148 +31,219 @@ def test_session_retrieved_and_set_on_context(self, _, mock_DAFNISession):
+
         # SETUP
-        session = MagicMock()
-        mock_DAFNISession.return_value = session
-        runner = CliRunner()
-        metadata = MagicMock()
-        metadata.dataset_id = "dataset-id"
-        metadata.files = []
-        mock_parse_dataset_metadata.return_value = metadata
+        self.metadata.files = []
 
         # CALL
-        result = runner.invoke(download.download, ["dataset", "version-id"])
+        result = self._run_command(directory=None, files=None)
 
         # ASSERT
-        mock_DAFNISession.assert_called_once()
-        mock_cli_get_latest_dataset_metadata.assert_called_once_with(
-            session, "version-id"
+        self.mock_DAFNISession.assert_called_once()
+        self.mock_cli_get_latest_dataset_metadata.assert_called_once_with(
+            self.mock_session, self.version_id
         )
-        mock_parse_dataset_metadata.assert_called_once_with(
-            mock_cli_get_latest_dataset_metadata.return_value
+        self.mock_parse_dataset_metadata.assert_called_once_with(
+            self.mock_cli_get_latest_dataset_metadata.return_value
         )
-        mock_download_dataset.assert_not_called()
+        self.mock_cli_select_dataset_files.assert_not_called()
+        self.mock_download_dataset.assert_not_called()
 
         self.assertEqual(
             result.output,
             "There are no files currently associated with the Dataset to download\n",
         )
 
         self.assertEqual(result.exit_code, 0)
+
+    def test_download_dataset_with_specific_files_given(
+        self,
+    ):
+        """Tests that the 'download dataset' command works correctly (when given
+        a list of files match to)"""
+
+        # CALL
+        files = ["filename1.zip", "filename2.zip", "*.csv"]
+        result = self._run_command(directory=None, files=files)
+
+        # ASSERT
+        self.mock_DAFNISession.assert_called_once()
+        self.mock_cli_get_latest_dataset_metadata.assert_called_once_with(
+            self.mock_session, self.version_id
+        )
+        self.mock_parse_dataset_metadata.assert_called_once_with(
+            self.mock_cli_get_latest_dataset_metadata.return_value
+        )
+        self.mock_cli_select_dataset_files.assert_called_once_with(
+            self.metadata, files=tuple(files)
+        )
+
+        self.mock_download_dataset.assert_called_once_with(
+            self.mock_session,
+            self.selected_dataset_files,
+            None,
+        )
+
+        self.assertEqual(result.exit_code, 0)
+
+    def test_download_dataset_with_no_selected_files(
+        self,
+    ):
+        """Tests that the 'download dataset' command works correctly when
+        there are no selected files to download"""
+
+        # SETUP
+        self.selected_dataset_files = []
+        self.mock_cli_select_dataset_files.return_value = self.selected_dataset_files
+
+        # CALL
+        result = self._run_command(directory=None, files=None)
```

Comment:
Maybe I've just gotten myself confused here, but don't we want to be passing in some files here (just so that the test scenario makes sense)?